config/v1/types_build.go

```diff
@@ -87,6 +91,27 @@ type ProxyConfig struct {
 	NoProxy *string `json:"noProxy,omitempty"`
 }
 
+type RegistriesList []string
+
+type RegistriesConfig struct {
+	// SearchRegistries lists the registries to search for images if an image repository is not specified in an image pull spec
+	// +optional
+	SearchRegistries RegistriesList `json:"searchRegistries,omitempty"`
+	// InsecureRegistries are registries which do not have a valid SSL certificate or only support HTTP connections
+	// +optional
+	InsecureRegistries RegistriesList `json:"insecureRegistries,omitempty"`
+	// BlockedRegistries are blacklisted from image pull/push. All other registries are allowed.
+	//
+	// Only one of BlockedRegistries or AllowedRegistries may be set.
+	// +optional
+	BlockedRegistries RegistriesList `json:"blockedRegistries,omitempty"`
+	// AllowedRegistries are whitelisted for image pull/push. All other registries are blocked.
+	//
+	// Only one of BlockedRegistries or AllowedRegsitries may be set.
+	// +optional
+	AllowedRegistries RegistriesList `json:"allowedRegistries,omitempty"`
```

Comment:
Approach will utilize the container runtime's `policy.json` and `registry.conf` as follows: https://github.com/containers/image/issues/548#issuecomment-452411961